File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum =0;
        int max = Integer.MIN_VALUE;
        for(int e : nums){
            sum+=e;
            max = Math.max(sum,max);
            if(sum<0) sum =0;
        }
        return max;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let sum =0;
    let max =  Number.NEGATIVE_INFINITY;
    
    for(let value of nums){
        sum +=value;
        max= Math.max(max, sum);
        if(sum<0) sum = 0;
    }
    return max;
};